evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1743,6 +1750,13 @@ where
             builder.add_gate(NoopGate, vec![]);
         }
 
+        // When using test configurations, the block circuit's degree is less than the
+        // 2-to-1 circuit's degree. Therefore, we also need to ensure its size meets
+        // the 2-to-1 circuit's recursion threshold degree bits.
+        while log2_ceil(builder.num_gates()) < TWO_TO_ONE_BLOCK_CIRCUIT_TEST_THRESHOLD_DEGREE_BITS {
```

Comment:
What if we want to test another circuit that doesn't require the same number of degree bits? I don't think `TWO_TO_ONE_BLOCK_CIRCUIT_TEST_THRESHOLD_DEGREE_BITS` should be constant, because even though now, we're only using this to test the 2-to-1 circuit in the CI, we should still be able to use this for other circuits that might not have the same size...

Also, I'm actually surprised `TWO_TO_ONE_BLOCK_CIRCUIT_TEST_THRESHOLD_DEGREE_BITS` is greater than `common.degree_bits()`. Do you know why?

Reply:
`TWO_TO_ONE_BLOCK_CIRCUIT_TEST_THRESHOLD_DEGREE_BITS` is the **smallest** degree bits we can get with a low-security STARK config for the `2-1-block circuit`. We will not use this value for other circuits.

We need it in the `block wrapper circuit` because we need to match its degree with the `2-1-block circuit`’s degree, but we build this circuit before the `2-1-block circuit` so we need to know this value in advance.

Reply:
But why does the common circuit data not contain a high enough degree then? Shouldn't the common data and the circuit we're building agree on the `degree_bits`?
Also, if we keep this, maybe we should have something like:
`while log2_ceil(builder.num_gates()) < max(TWO_TO_ONE_BLOCK_CIRCUIT_TEST_THRESHOLD_DEGREE_BITS, block.circuit.common.degree_bits())`
to not have two different `while` loops doing the same thing?

Reply:
The previous check `log2_ceil(builder.num_gates()) < block.circuit.common.degree_bits()` is a hack, as described in the comments.

The `degree_bits` is needed for this circuit is determined in the next layer of recursion, while the common data is from the  last layer of recursion. Therefore, they may not align. We could use a trial-and-error loop to find this number, but it would be much slower and seems excessive for now.

Reply:
I’ve completed the refactoring to keep only one `while`. PTAL